sycl/test/basic_tests/is_device_copyable.cpp

```diff
@@ -0,0 +1,70 @@
+// RUN: not %clangxx -fsycl -fsycl-device-only -fsyntax-only -Xclang -verify \
+// RUN: %s -I %sycl_include 2>&1 | FileCheck -check-prefix=CHECK-ERR %s
+
+// This test checks if compiler reports compilation error on an attempt to pass
+// a struct with type that is not device copyable as SYCL kernel parameter.
+// It also checks that the error is not emitted if the type trait
+// is_device_copyable is specialized for such types as true_type.
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+struct A { int i; };
+
+struct B {
+  int i;
+  B (int _i) : i(_i) {}
+  B (const B& x) : i(x.i) {}
+};
+
+// Identical to B, but will be declared as device copyable.
+struct BCopyable {
+  int i;
+  BCopyable (int _i) : i(_i) {}
+  BCopyable (const BCopyable& x) : i(x.i) {}
+};
+
+struct C : A {
+  const A C2;
+  C() : A{0}, C2{2}{}
+};
+
+struct D {
+  int i;
+  ~D();
+};
+
+// Identical to D, but will be declared as device copyable.
+struct DCopyable {
+  int i;
+  ~DCopyable();
+};
+
```

Comment:
This is a good start to testing, but only tests some of the super common stuff. Please add tests for *all* of the ways that checking device copyability can fail (like base classes), as well as edge case scenarios (classes with no members or bases, virtual bases with difficult inheritance patterns, classes with noncopyable static members (as those don't get copied as part of the object), bit-fields, and whatever other odd situations you can come up with.

Reply:
This test is rework of the deleted clang test that is deleted in this PR: https://github.com/intel/llvm/pull/3905/files#diff-75331d55b17d64c7f82af56c111baced01571af4ef7eef17678e3bb730e2dc08
I'll add more tests for sure.